arraycontext/container/__init__.py

```diff
@@ -115,6 +117,10 @@ class ArrayContainer:
     """
 
 
+class ArrayContainerTypeError(TypeError):
```

Comment:
@inducer Added this in 3499359. Not sold on the name, but now it should be easy to `sed` for it.